lib/SimpleSAML/Logger.php

```diff
@@ -13,9 +13,14 @@
 class Logger
 {
     /**
-     * @var \SimpleSAML\Logger\LoggingHandlerInterface|false|null
+     * @var \SimpleSAML\Logger\LoggingHandlerInterface
      */
-    private static $loggingHandler = null;
+    private static $loggingHandler;
+
+    /**
+     * @var bool
+     */
+    private static $initializing = false;
```

Comment:
By introducing a new property, we can keep the typedecl of `$loggingHandler` as simple as possible..